### PR TITLE
Session tests

### PR DESCRIPTION
**Pull request by zhukovra**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #105

### CODE REVIEW

File: tests/Session/SessionTest.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        @session_destroy();
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testGetAndSet()
+    {
+        $session = new Session();
+        $session->set('get', 'set');
```
Comment (samdark) [inline tests/Session/SessionTest.php]: Let's make these names a bit more obvious.

```suggestion
        $session->set('key_get', 'value');
```

File: tests/Session/SessionTest.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        self::assertEquals('set', $session->get('get'));
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        self::assertEquals('value', $session->get('key_get'));
```

File: tests/Session/SessionTest.php [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testHas()
+    {
+        $session = new Session();
+        $session->set('has', 'has');
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        $session->set('key_has', 'value');
```

File: tests/Session/SessionTest.php [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        self::assertTrue($session->has('has'));
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        self::assertTrue($session->has('key_has'));
```

File: tests/Session/SessionTest.php [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testClose()
+    {
+        $session = new Session();
+        $session->set('close', 'close');
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        $session->set('key_close', 'value');
```

File: tests/Session/SessionTest.php [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        $session->close();
+        self::assertEquals(PHP_SESSION_NONE, session_status());
+        // because session_destroy() in tearDown doesn't work as expected
```
Comment (samdark) [inline tests/Session/SessionTest.php]: Should we instead make it work as expected?

File: tests/Session/SessionTest.php [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        self::assertNotEquals($id, $session->getId());
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testDiscard()
+    {
+        $session = new Session();
+        $session->set('discard', 'discard');
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        $session->set('key_discard', 'value');
```

File: tests/Session/SessionTest.php [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+        $session->discard();
+        self::assertEmpty($session->get('discard'));
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        self::assertEmpty($session->get('key_discard'));
```

File: tests/Session/SessionTest.php [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+        self::assertEquals($session->getName(), session_name());
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testPull()
+    {
+        $session = new Session();
+        $session->set('pull', 'pull');
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        $session->set('key_pull', 'value');
```

File: tests/Session/SessionTest.php [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+        self::assertEquals('pull', $session->pull('pull'));
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        self::assertEquals('value', $session->pull('key_pull'));
```

File: tests/Session/SessionTest.php [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+        self::assertEmpty($session->get('pull'));
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        self::assertEmpty($session->get('key_pull'));
```

File: tests/Session/SessionTest.php [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    /**
+     * @expectedException \RuntimeException
+     */
+    public function testAlreadyStartedException()
+    {
+        $session = new Session();
+        $session->set('1', 1);
+        $session = new Session();
```
Comment (samdark) [inline tests/Session/SessionTest.php]:
```suggestion
        new Session();
```

File: tests/Session/SessionTest.php (hunk 6 above)
Comment (zhukovra) [reply tests/Session/SessionTest.php]: Yes. I run tests without separated processes. With separating we don't need that and tearDown() function

File: tests/Session/SessionTest.php (hunk 12 above)
Comment (zhukovra) [reply tests/Session/SessionTest.php]: We can slightly simple (with `$session->open()`).

**Comment by zhukovra** [comment]
@samdark all review notes was fixed

**Comment by samdark** [comment]
Thanks!

**Pull request merged**